### PR TITLE
Fixed #2777 and added aspect control in LineAnimator

### LINKED ISSUE #2777: LineAnimator display of small y-axis values
The aspect ratio of `LineAnimator` appears to be determined through absolute scaling of x and y values. For example if y ~ 10^-9 and x ~ 1, the y-axis is not visible. We need something like an `aspect` kwarg as in `plt.imshow` so that the length of each axis is determined by the plotting area and not the data range. This issue probably applies to the other animated classes.

### PR DESCRIPTION
**Pull request by Akram9**
### Description

Fixes #2777 
Changed y-axis pad size in `base.py` to fixed size and added `aspect` option in `LineAnimator`.

### CODE REVIEW

**Comment by nabobalis** [comment]
What do the new plots look like now?

**Comment by Akram9** [comment]
new0
new2
new1

They don't fluctuate in y-axis height now.

**Comment by nabobalis** [comment]
What was it like before the fix?

**Comment by Akram9** [comment]
aspect_0 25-3
aspect_0 25-2
aspect_8-2
aspect_0 25-5

**Comment by Akram9** [comment]
The y-axis padding height was based on the input data. I changed that to fixed size padding.

File: sunpy/visualization/animator/line.py [hunk 1]
```diff
@@ -77,7 +78,7 @@ class LineAnimator(ArrayAnimator):
           In this scenario the same x-axis values are used in every frame of the animation.
        b) Second, the x-axis array can have the same shape as the data array.
           In this scenario the x-axis is refreshed for each frame. For example, if
-          ``data.shape == axis_ranges[plot_axis_index] == (4, 3)``,
+          ``data.shape == axis_ranges[plot_axis_index] == (4, 3)``,     #shouldn't it be axis_ranges[plot_axis_index].shape ?
```
Comment (nabobalis) [inline sunpy/visualization/animator/line.py]: Can you check this?

**Comment by nabobalis** [comment]
Looks much better than before.

File: sunpy/visualization/animator/line.py (hunk 1 above)
Comment (Akram9) [reply sunpy/visualization/animator/line.py]: Yeah, I was going to say, I forgot the comment there.

Comment (Akram9) [reply sunpy/visualization/animator/line.py]: If you don't mind, I'll check this about 14 hours later.? I'm a bit occupied now.

Comment (nabobalis) [reply sunpy/visualization/animator/line.py]: Sure, there is no rush.

**Comment by nabobalis** [comment]
I wonder if a new figure test is what we want here.

**Comment by Akram9** [comment]
I will see what I can do here.

File: sunpy/visualization/animator/line.py (hunk 1 above)
Comment (Akram9) [reply sunpy/visualization/animator/line.py]: Yes, it should be as I expect it.

**Comment by Akram9** [comment]
I am new to tests. Should it be a unit test comparing data shapes?

**Comment by Akram9** [comment]
Okay, I got a problem here. Changing the window size destroys the look as the pad size is fixed. I am trying a fix.

**Comment by nabobalis** [comment]
I am not sure how best to test this behavior, if it can be done at all. Maybe the axes size of a plot? Maybe a figure test?

**Comment by Akram9** [comment]
Yes, axes size of the plot to figure size is what I had in mind.

**Comment by Cadair** [comment]
If I had to guess looking at your diff I would think that you don't need the changes in `base.py` for this to work?

**Comment by Akram9** [comment]
I had thought the same and then I made changes in `base` which sort of solved the problem so I had gone ahead with that. 
Anyway, I will keep it unchanged and try only from `line`.

**Comment by Akram9** [comment]
Applied a fix which fixes the aforementioned issues but I have only tried the example as in the docs. I will say it is yet not complete.

File: sunpy/visualization/animator/base.py [hunk 2]
```diff
@@ -285,7 +285,7 @@ def _add_widgets(self):
             x = i*2
             # The i+1/10. is a bug that if you make two axes directly on top of
             # one another then the divider doesn't work.
-            self.buttons.append(self.fig.add_axes((0., 0., 0.+i/10., 1.)))
+            self.buttons.append(self.fig.add_axs((0., 0., 0.+i/10., 1.)))
```
Comment (DanRyanIrish) [inline sunpy/visualization/animator/base.py]: Is this a typo of a correct change?

**Comment by DanRyanIrish** [comment]
I think this PR now needs a ```git pull upstream master```.

File: sunpy/visualization/animator/base.py (hunk 2 above)
Comment (Akram9) [reply sunpy/visualization/animator/base.py]: Oops, a typo.

**Pull request closed without merging**